### PR TITLE
rustdoc: Don't counts ids twice when using --enable-commonmark

### PR DESCRIPTION
**Pull request by ollie27**
cc @GuillaumeGomez
r? @QuietMisdreavus

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Oh awesome! I think the checks will fail because of one test I updated to make the original html-diff PR get merged though. If not, it means the PR is incomplete.

**Comment by QuietMisdreavus** [comment]
In terms of "reducing false-positive render difference warnings", this has the same effect as #44350. What test are you thinking of that would be affected by this?

**Comment by ollie27** [comment]
I think the test change you're referring to was reverted in https://github.com/rust-lang/rust/pull/44238/files#diff-f6ee607baf038ab9752a5e4a57e92d4d so there's no problem.

**Comment by QuietMisdreavus** [comment]
This passed travis, that's good enough for me. `:D` See https://github.com/rust-lang/rust/pull/44350#issuecomment-327560425 for where this leaves that other PR.

@bors r+

**Comment by GuillaumeGomez** [comment]
@ollie27: Perfect then!

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -1752,9 +1752,13 @@ fn render_markdown(w: &mut fmt::Formatter,
                    prefix: &str,
                    scx: &SharedContext)
                    -> fmt::Result {
-    let hoedown_output = format!("{}", Markdown(md_text, RenderType::Hoedown));
     // We only emit warnings if the user has opted-in to Pulldown rendering.
     let output = if render_type == RenderType::Pulldown {
+        // Save the state of USED_ID_MAP so it only gets updated once even
+        // though we're rendering twice.
+        let orig_used_id_map = USED_ID_MAP.with(|map| map.borrow().clone());
+        let hoedown_output = format!("{}", Markdown(md_text, RenderType::Hoedown));
```
Comment (frewsxcv) [inline src/librustdoc/html/render.rs]: don't need to change it, but in the future, using `.to_string()` is the same as `format!("{}", ..)`

**Pull request merged**